### PR TITLE
NCL-7432 Add OpenTelemetry instrumentation to GME

### PR DESCRIPTION
**Pull request by rnc**

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -192,16 +192,17 @@ subprojects {
...
+    extra["ivyVersion"] = "2.5.0"
     extra["jacksonVersion"] = "2.11.2"
+    extra["jgitVersion"] = "6.3.0.202209071007-r"
     extra["junitVersion"] = "4.13.1"
     extra["logbackVersion"] = "1.2.9"
     extra["mavenVersion"] = "3.5.0"
+    extra["opentelemetryVersion"] = "1.2.0"
     extra["ownerVersion"] = "1.0.12"
-    extra["pmeVersion"] = "4.8"
+    extra["pmeVersion"] = "4.14-SNAPSHOT"
```
Comment (rnc) [inline build.gradle.kts:203]: Need to release PME 4.14 first

File: common/src/main/java/org/jboss/gm/common/utils/OTELUtils.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+            } finally {
+                FilteringCustomLogger.getContext().setLevel(originalLevel);
+            }
+        }
+    }
+
+    public void stopOTel() {
+        try {
+            if (originalLevel != LogLevel.DEBUG) {
+                FilteringCustomLogger.getContext().setLevel(LogLevel.DEBUG);
```
Comment (dwalluck) [inline common/src/main/java/org/jboss/gm/common/utils/OTELUtils.java:44]: Should be `originalLevel`?

File: common/src/main/java/org/jboss/gm/common/utils/OTELUtils.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        if (endpoint != null) {
+            if (service == null) {
+                service = "gradle-manipulator";
+            }
+            logger.info("Enabling OpenTelemetry collection on {} with service name {} level {}", endpoint, service,
+                    originalLevel);
+            try {
+                if (originalLevel != LogLevel.DEBUG) {
+                    FilteringCustomLogger.getContext().setLevel(LogLevel.DEBUG);
+                }
```
Comment (dwalluck) [inline common/src/main/java/org/jboss/gm/common/utils/OTELUtils.java:32]: The log message seems a little confusing to me as it prints `level originalLevel` in the log, but actually forces it to use `DEBUG`?

File: common/src/main/java/org/jboss/gm/common/utils/OTELUtils.java (hunk 2 above)
Comment (dwalluck) [reply common/src/main/java/org/jboss/gm/common/utils/OTELUtils.java:44]: Eh, no, you do set that in the finally. But, it should have already been set on start, line 36, so I wasn't sure.

File: common/src/main/java/org/jboss/gm/common/utils/OTELUtils.java (hunk 3 above)
Comment (rnc) [reply common/src/main/java/org/jboss/gm/common/utils/OTELUtils.java:32]: I was trying to always get the output from the opentelemetry-java-cli library as its useful to see the trace and span ids.

File: common/src/main/java/org/jboss/gm/common/utils/OTELUtils.java (hunk 2 above)
Comment (rnc) [reply common/src/main/java/org/jboss/gm/common/utils/OTELUtils.java:44]: In each start/stop there is a wrapper around the logging control.

File: common/src/main/java/org/jboss/gm/common/utils/OTELUtils.java (hunk 3 above)
Comment (rnc) [reply common/src/main/java/org/jboss/gm/common/utils/OTELUtils.java]: I've removed the confusing bit of the logging statement (left over debug that I missed)

**Review by dwalluck — APPROVED** [review]

**Pull request merged**